mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -45,45 +64,53 @@ func runIgnitionFailure(c cluster.TestCluster) {
 	}
 }
 
-func ignitionFailure(c cluster.TestCluster) error {
-	// We can't create files in / due to the immutable bit OSTree creates, so
-	// this is a convenient way to test Ignition failure.
-	failConfig, err := conf.EmptyIgnition().Render(conf.FailWarnings)
-	if err != nil {
-		return errors.Wrapf(err, "creating empty config")
+func runDualBootfsFailure(c cluster.TestCluster) {
+	if err := dualBootfsFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	failConfig.AddFile("/notwritable.txt", "Hello world", 0644)
+}
 
-	builder := platform.NewQemuBuilder()
-	defer builder.Close()
-	builder.SetConfig(failConfig)
-	err = builder.AddBootDisk(&platform.Disk{
-		BackingFile: kola.QEMUOptions.DiskImage,
-	})
-	if err != nil {
-		return err
+func runDualBootfsIgnitionFailure(c cluster.TestCluster) {
+	if err := dualBootfsIgnitionFailure(c); err != nil {
+		c.Fatal(err.Error())
 	}
-	builder.MemoryMiB = 1024
-	builder.Firmware = kola.QEMUOptions.Firmware
+}
+
+// take string and grep for it in console logs
+func VerifyError(builder *platform.QemuBuilder, tempLogFile string, searchString string) error {
 	inst, err := builder.Exec()
 	if err != nil {
 		return err
 	}
 	defer inst.Destroy()
-
 	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+
 	defer cancel()
 
 	errchan := make(chan error)
 	go func() {
-		err := inst.WaitAll(ctx)
-		if err == nil {
-			err = fmt.Errorf("Ignition unexpectedly succeeded")
-		} else if err == platform.ErrInitramfsEmergency {
-			// The expected case
-			err = nil
+		resultingError := inst.WaitAll(ctx)
+		if resultingError == nil {
+			resultingError = fmt.Errorf("ignition unexpectedly succeeded")
+		} else if resultingError == platform.ErrInitramfsEmergency {
+			// Expectred initramfs failure, checking the console file to insure
+			// that coreos.ignition.failure failed
+			b, err := os.ReadFile(tempLogFile)
+			if err != nil {
+				resultingError = err
```

Comment:
This is going to continue execution. A cleaner approach is to make the opening and grepping a separate function. Then here you can do something like

```golang
} else if resultingError == platform.ErrInitramfsEmergency {
	if found, err := fileHasPattern(tempLogFile, pattern); err != nil {
		resultingError = errors.Wrapf(err, ...)
	} else if !found {
		resultingError = fmt.Errorf(...)
	} else {
		// The expected case
		resultingError = nil
	}
} ...
```

Reply:
Just for future reference - tempLogFile was not being used by the builder.
Fix:
https://github.com/coreos/coreos-assembler/pull/3647/files#diff-3ad99cf50017bf15cd0eae37f7571c6e217480c1c2b1748266bc7eb417cb0aacR166-R167